demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -0,0 +1,385 @@
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    Field,
+    FieldSet,
+    FinalForm,
+    FinalFormInput,
+    FinalFormSaveSplitButton,
+    FinalFormSubmitEvent,
+    Loading,
+    MainContent,
+    TextField,
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    ToolbarTitleItem,
+    useFormApiRef,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { ArrowLeft } from "@comet/admin-icons";
+import { EditPageLayout, queryUpdatedAt, resolveHasSaveConflict, useFormSaveConflict } from "@comet/cms-admin";
+import { IconButton } from "@mui/material";
+import { FormApi } from "final-form";
+import { filter } from "graphql-anywhere";
+import isEqual from "lodash.isequal";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { createManufacturerMutation, manufacturerFormFragment, manufacturerQuery, updateManufacturerMutation } from "./ManufacturerForm.gql";
+import {
+    GQLCreateManufacturerMutation,
+    GQLCreateManufacturerMutationVariables,
+    GQLManufacturerFormDetailsFragment,
+    GQLManufacturerQuery,
+    GQLManufacturerQueryVariables,
+    GQLUpdateManufacturerMutation,
+    GQLUpdateManufacturerMutationVariables,
+} from "./ManufacturerForm.gql.generated";
+
+type FormValues = Omit<GQLManufacturerFormDetailsFragment, "address" | "addressAsEmbeddable"> & {
+    address:
+        | (Omit<NonNullable<GQLManufacturerFormDetailsFragment["address"]>, "streetNumber" | "zip" | "alternativeAddress"> & {
```

Comment:
why NonNullable?

Reply:
because address is defined as `{street: string, ....} | null` and for Omit to work null has to be removed. same for alternativeAddress with `NonNullable<NonNullable<TYPE["address"]>["alternativeAddress"]>` (took me some time to find the reason for typescript to complain about input not compatible with FormValues in update-mutation :-D)